### PR TITLE
[docs, API] control api + swagger ui for http api

### PR DESCRIPTION
**Pull request by balopat**
**Description**

This became pretty massive just from writing the Control API doc: 

- Control API docs 
- Swagger UI for HTTP 
- markdown template for the gRPC API 
- removes /v1/events as it is superfluous 
- removes /v1/handle as it was a remainder of our plugins experiment - we can reintroduce it if necessary later
- removes `stream LogEntry` as a request parameter from `EventLog` because it is completely unnecessary 

**User facing changes**

simpler API - no  /v1/events, just  /v1/event_log

**Submitter Checklist**

These are the criteria that every PR should meet, please check them off as you
review them:

- ~[ ] Includes [unit tests](../DEVELOPMENT.md#creating-a-pr)~
- [x] Mentions any output changes.
- [x] Adds documentation as needed: user docs, YAML reference, CLI reference.
- ~[ ] Adds integration tests if needed.~

**Reviewer Notes**

- [ ] The code flow looks good. 
- [ ] Unit test added.
- [ ] User facing changes look good.

### CODE REVIEW

**Comment by balopat** [comment]
- [ ] figure out filtering out methods 
- [ ] fix CLI menupoint weight
- [ ] revert /v1/handle and hide it & mark it as experimental
- [ ] revert /v1/events_log and hide it & mark it as deprecated
- [ ] remove /v1/events 
- [ ] rebase

File: docs/content/en/docs/design/api.md [hunk 1]
```diff
@@ -14,11 +14,11 @@ pipeline** and for **controlling the phases in the pipeline**.
 
 To retrieve information about the Skaffold pipeline, the Skaffold API provides two main functionalities:
   
-  * A [streaming event log]({{< relref "/docs/design/api#events-api">}}) created from the different phases in a pipeline run and
+  * A [streaming event log]({{< relref "#events-api">}}) created from the different phases in a pipeline run and
```
Comment (nkubala) [inline docs/content/en/docs/design/api.md]:
```suggestion
  * A [streaming event log]({{< relref "#events-api">}}) created from the different phases in a pipeline run, and
```

File: docs/content/en/docs/design/api.md [hunk 2]
```diff
@@ -14,11 +14,11 @@ pipeline** and for **controlling the phases in the pipeline**.
...
   
-  * A snapshot of the [overall state]({{< relref "/docs/design/api#state-api" >}}) of the pipeline at any given time during the run.
+  * A snapshot of the [overall state]({{< relref "#state-api" >}}) of the pipeline at any given time during the run.
 
-To control the individual phases of the Skaffold, the Skaffold API provides [fine grained control over]({{< relref "/docs/design/api#controlling-build-sync-deploy" >}})
+To control the individual phases of the Skaffold, the Skaffold API provides [fine grained control over]({{< relref "#controlling-build-sync-deploy" >}})
```
Comment (nkubala) [inline docs/content/en/docs/design/api.md]:
```suggestion
To control the individual phases of the Skaffold, the Skaffold API provides [fine grained control]({{< relref "#controlling-build-sync-deploy" >}}) over
```

File: docs/content/en/docs/design/api.md [hunk 3]
```diff
@@ -27,7 +27,7 @@ The Skaffold API is `gRPC` based, and it is also exposed via the gRPC gateway as
 The server is hosted locally on the same host where the skaffold process is running, and will serve by default on ports 50051 and 50052.
 These ports can be configured through the `--rpc-port` and `--rpc-http-port` flags.
 
-The server's gRPC service definitions and message protos can be found [here]({{< relref "/docs/references/api" >}}).
+We generate the server's [gRPC service definitions and message protos]({{< relref "/docs/references/api/grpc" >}}) as well as the [Swagger based HTTP API Spec]({{< relref "/docs/references/api/swagger" >}}).
```
Comment (nkubala) [inline docs/content/en/docs/design/api.md]:
```suggestion
For reference, we generate the server's [gRPC service definitions and message protos]({{< relref "/docs/references/api/grpc" >}}) as well as the [Swagger based HTTP API Spec]({{< relref "/docs/references/api/swagger" >}}).
```

File: docs/content/en/docs/design/api.md [hunk 4]
```diff
@@ -123,15 +123,15 @@ Using `curl` and `HTTP_RPC_PORT=50052`, an example output of a `skaffold dev` ex
 ```
 {{% /tab %}}
 {{% tab "gRPC API" %}}
-To get events from the `gRPC` server, first create [`gRPC` client]({{< relref "/docs/design/api#creating-a-grpc-client" >}})
+To get events from the `gRPC` server, first create [`gRPC` client]({{< relref "#creating-a-grpc-client" >}})
```
Comment (nkubala) [inline docs/content/en/docs/design/api.md]:
```suggestion
To get events from the API using `gRPC`, first create a [`gRPC` client]({{< relref "#creating-a-grpc-client" >}}):
```

File: docs/content/en/docs/design/api.md [hunk 5]
```diff
@@ -123,15 +123,15 @@ Using `curl` and `HTTP_RPC_PORT=50052`, an example output of a `skaffold dev` ex
...
 
 ```golang
 func main() {
   ctx, ctxCancel := context.WithCancel(context.Background())
   defer ctxCancel()
   // `client` is the gRPC client with connection to localhost:50051.
   // See code above to create it
-  logStream, err := client.Events(ctx)
+  logStream, err := client.EventLog(ctx, &empty.Empty{})
```
Comment (nkubala) [inline docs/content/en/docs/design/api.md]: didn't we want to keep this as `client.Events`?

File: docs/content/en/docs/design/api.md [hunk 6]
```diff
@@ -150,7 +150,7 @@ func main() {
 {{% /tab %}}
 {{% /tabs %}}
 
-Each [Entry log]({{<relref "/docs/references/api#proto.LogEntry" >}}) contains an [Event]({{< relref "/docs/references/api#proto.Event" >}}) in the `LogEntry.Event` field and
+Each [Entry log]({{<relref "/docs/references/api/grpc#proto.LogEntry" >}}) contains an [Event]({{< relref "/docs/references/api/grpc#proto.Event" >}}) in the `LogEntry.Event` field and
```
Comment (nkubala) [inline docs/content/en/docs/design/api.md]:
```suggestion
Each [Entry]({{<relref "/docs/references/api/grpc#proto.LogEntry" >}}) in the log contains an [Event]({{< relref "/docs/references/api/grpc#proto.Event" >}}) in the `LogEntry.Event` field and
```

File: docs/content/en/docs/design/api.md [hunk 7]
```diff
@@ -214,7 +214,7 @@ Using `curl` and `HTTP_RPC_PORT=50052`, an example output of a `skaffold dev` ex
 ```
 {{% /tab %}}
 {{% tab "gRPC API" %}}
-To get events over `gRPC` server, first create [`gRPC` client]({{< relref "/docs/design/api#creating-a-grpc-client" >}})
+To get events over `gRPC` server, first create [`gRPC` client]({{< relref "#creating-a-grpc-client" >}})
```
Comment (nkubala) [inline docs/content/en/docs/design/api.md]:
```suggestion
To retrieve the state from the server using `gRPC`, first create [`gRPC` client]({{< relref "#creating-a-grpc-client" >}}) as before:
```

File: docs/content/en/docs/design/api.md (hunk 2 above)
Comment (balopat) [reply docs/content/en/docs/design/api.md]: I don't get this change

File: docs/content/en/docs/design/api.md (hunk 5 above)
Comment (balopat) [reply docs/content/en/docs/design/api.md]: yes we should, good catch

Comment (balopat) [reply docs/content/en/docs/design/api.md]:
```suggestion
  logStream, err := client.Events(ctx, &empty.Empty{})
```

File: docs/content/en/docs/design/api.md (hunk 2 above)
Comment (tejal29) [reply docs/content/en/docs/design/api.md]: this is very small nit. i am going to approve and merge

**Pull request merged**